File: Modulo 2 - Arrays/Exercicio0/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

// Calcula o nível aproximado necessário para compreender algum texto e retorna a grade
int main(void)
{
    string s = get_string("Texto: ");
    int letras = 0;
    int palavras = 1;
    int frases = 0;

    // Calcula a quantidade de letras, palavras e frases
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letras++;
        }
        else if (s[i] == ' ')
        {
            palavras++;
        }
        else if ((s[i] == '.') || (s[i] == '!') || (s[i] == '?'))
        {
            frases++;
        }
    }

    float media_letras_por_palavras = ((float)letras / palavras) * 100;
    float media_frases_por_palavras = ((float)frases / palavras) * 100;

    // Indice Coleman Liau:
    int result = round((0.0588 * media_letras_por_palavras) - (0.296 * media_frases_por_palavras) - 15.8);

    if (result >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", result);
    }
}

File: Modulo 6 - Python/exercicio5/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0;
words = 1;
phrases = 0;

for i in text:

    # Calcula a quantidade de letras
    if i.isalpha() or i.isnumeric():
        letters +=1

    # Calcula a quantidade de palavras
    elif i == ' ':
        words +=1

    # Calcula a quantidade de frases
    elif i == '.' or i == '!' or i == '?':
        phrases += 1

average_letters_by_words = (letters / words) * 100
average_phrases_by_words = (phrases / words) * 100

result = round((0.0588 * average_letters_by_words) - (0.296 * average_phrases_by_words)  - 15.8)

if result >= 16:
    print("Grade 16+")
elif result < 1:
    print("Before Grade 1")
else:
    print(f"Grade {result}")